### ISSUE TITLE
Native Teams Templates support - provisioning

### ISSUE THREAD

**Issue by alexc-MSFT** [OfficeDev/microsoft-teams-apps-requestateam#200]
**Summary**
With this release the app template will be able to provision templates created through the admin center.

**Intended Outcome**
To be able to integrate with the new native templates functionality.

**How will it work**
You will be able to define templates in the Teams Admin Centre.

You will then copy the ID of the template and paste it into the 'Teams Templates' list.

The templates list will be simplified to remove 'First Party Template', 'Team ID' and any other columns that are no longer required.

We will also remove the functionality that 'clones' teams from the Logic App to simplify this.

Finally, the documentation will be updated.

**Comment by alexc-MSFT** [comment]
We are currently testing this and it appears to work 👍

**Comment by adm-mp76** [comment]
When will it be released?

**Comment by alexc-MSFT** [comment]
@adm-mp76 technically today, it's just been approved and is live in the master branch.

I am going to write up documentation next week on how to use it 👍

**Comment by adm-mp76** [comment]
This is awesome, works great!
We are using the old version and you suggest upgrading to newer to support upcoming Naming Conventions functionality
How do we upgrade?.....what parts/files needs update? (maybe a guide for this would be great)
Thanks!

**Comment by jcmmarq** [comment]
> **Summary**
> With this release the app template will be able to provision templates created through the admin center.
> 
> **Intended Outcome**
> To be able to integrate with the new native templates functionality.
> 
> **How will it work**
> You will be able to define templates in the Teams Admin Centre.
> 
> You will then copy the ID of the template and paste it into the 'Teams Templates' list.
> 
> The templates list will be simplified to remove 'First Party Template', 'Team ID' and any other columns that are no longer required.
> 
> We will also remove the functionality that 'clones' teams from the Logic App to simplify this.
> 
> Finally, the documentation will be updated.

Do you have an ETA or is the a current way to do it with the current version?

**Comment by adm-mp76** [comment]
> > **Summary**
> > With this release the app template will be able to provision templates created through the admin center.
> > **Intended Outcome**
> > To be able to integrate with the new native templates functionality.
> > **How will it work**
> > You will be able to define templates in the Teams Admin Centre.
> > You will then copy the ID of the template and paste it into the 'Teams Templates' list.
> > The templates list will be simplified to remove 'First Party Template', 'Team ID' and any other columns that are no longer required.
> > We will also remove the functionality that 'clones' teams from the Logic App to simplify this.
> > Finally, the documentation will be updated.
> 
> Do you have an ETA or is the a current way to do it with the current version?

Previous versions of Request-a-team
If you are using a previous version of Request-a-team and wish to use the Admin Center templates, please follow the steps below:
Step 1: Define the Template in the Admin Center
Navigate to the Teams Admin Center.
Select 'Team templates' from the left hand menu (under 'Teams').
Click the '+Add' button to create a new template.
Choose whether to create a brand new template, use an existing team or start with an existing template.
Fill out the details and click 'Next' (Note - only English (United States) is supported with Request-a-team at the moment) 
Create channels and configure apps for the template and click 'Submit'.
Once created, click on the template and copy the 'Template ID' value.
Step 2: Add the template to the Request-a-team templates list
_Note - After the above steps have been completed please WAIT 24 hours before adding the template to the SharePoint list. Failure to wait 24 hours means provisioning may fail and if the template is added to the list then users of Request-a-team can use it. _
Navigate to the SharePoint site used by Request-a-team.
Open the 'Teams Templates' list.
Create a new list item with the following values:
Title - Title of the template you created in the Admin Center (does not have to match)
Description - A description for the template e.g. the long description specified when creating the template
Base Template Id - The Id for the template which you copied earlier
Base Template Type - Leave blank
Team Id - Leave blank
First Party Template - Yes
The template should now be ready for use.
We do recommend upgrading to the latest version if possible as this will be a prerequisite for the upcoming Naming Conventions functionality - Naming Conventions - Single.

**Comment by adm-mp76** [comment]
> @adm-mp76 technically today, it's just been approved and is live in the master branch.
> 
> I am going to write up documentation next week on how to use it 👍

Please Alex......How do we upgrade if we have already installed the old solution?

**Comment by alexc-MSFT** [comment]
@adm-mp76 The upgrade process is a bit tricky, we are trying to work on a way to simplify this but due to the fact the solution uses so many components and we can't tell if customers have customised these, we can't just assume they haven't and deploy updates over the top.

The SharePoint site has been updated and so has the ProcessTeamRequest logic app.

To update the SPO site, you can simply make these changes manually by deploying the updated app to a separate environment, comparing the changes and adding the new fields.

For the Logic App, you can as above deploy to a new tenant so you can see the changes. You can then either make these manually or edit the production Logic App in code view and paste the code from the other tenant, you will however need to change some values.

I can perhaps work with you to find a way to get this updated for you on an unofficial capacity outside of my work hours. Please ping me an email - alex.clark@microsoft.com.

**Comment by alexc-MSFT** [comment]
@jcmmarq Yes you can do this in the current version. If you follow the Teams Templates documentation in the Wiki, it explains how to do it.

**Comment by adm-mp76** [comment]
> @adm-mp76 The upgrade process is a bit tricky, we are trying to work on a way to simplify this but due to the fact the solution uses so many components and we can't tell if customers have customised these, we can't just assume they haven't and deploy updates over the top.
> 
> The SharePoint site has been updated and so has the ProcessTeamRequest logic app.
> 
> To update the SPO site, you can simply make these changes manually by deploying the updated app to a separate environment, comparing the changes and adding the new fields.
> 
> For the Logic App, you can as above deploy to a new tenant so you can see the changes. You can then either make these manually or edit the production Logic App in code view and paste the code from the other tenant, you will however need to change some values.
> 
> I can perhaps work with you to find a way to get this updated for you on an unofficial capacity outside of my work hours. Please ping me an email - [alex.clark@microsoft.com](mailto:alex.clark@microsoft.com).

Thanks Alex!
I believe we will wait for the next update (we need to be able to add suffix in form of projectnumber for our team)
Do you have an ETA of next update.....will it be jan or mars?
